.github/workflows/linkcheck.yml

```diff
@@ -0,0 +1,33 @@
+name: linkcheck in CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+  workflow_dispatch:
+    inputs:
+      failOnError:
+        description: 'Fail job on link check error'
+        required: false
+        default: 'false'
```

Comment:
Thanks for the constructive workflow_dispatch suggestion here for readability of continue-on-error below.